### PR TITLE
Updated rhea dependancy to 0.3.6 related to #352

### PR DESCRIPTION
**Pull request by anthonyvercolano**
Carry over #450 to put through the gate.

# Checklist
- [X] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
- [X] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)
*Messages not delivered and no exceptions thrown* https://github.com/Azure/azure-iot-sdk-node/issues/352
# Description of the problem
In summary, there is a *size* field which is initially set to 2048 when the device or twin is created. The value of the field is decreasing as events are sent and increased upon acknowledgement from the server. -If device twin is used, this value is not increased, and when it reaches 0 the device fails to submit messages (but is still connected, so no errors are raised). If device twin is not used, there are not issues.

# Description of the solution
An update has been provided from the rhea repo (https://github.com/amqp/rhea/issues/171), and updates to the dependencies have been made.

### CODE REVIEW

**Comment by pierreca** [comment]
Windows build will fail if the package-lock.json file isn't updated as well
